renku/service/views/datasets.py

```diff
@@ -0,0 +1,325 @@
+                }
+            )
+
+        pwd = validate_project_id(user, project_id)
+        if not pwd:
+            return jsonify(
+                error={
+                    'code': INVALID_PARAMS_ERROR_CODE,
+                    'message': 'invalid project_id argument',
+                }
+            )
+
+        cwd = os.getcwd()
+        os.chdir(str(pwd))
+        try:
+            add_file([str(local_path)],
+                     dataset_name,
+                     use_external_storage=False)
+            if not repo_sync(pwd, remote_name):
+                return jsonify(
+                    error={
+                        'code': INTERNAL_FAILURE_ERROR_CODE,
+                        'message': 'remote name not found'
+                    }
+                )
+
+        except (Exception, BaseException) as e:
+            return jsonify(
+                error={
+                    'code': INTERNAL_FAILURE_ERROR_CODE,
+                    'message': 'failed to add file or sync with remote',
+                    'exception': str(e)
+                }
+            )
+
+        finally:
+            os.chdir(cwd)
+
+        return jsonify(result={
+            'dataset': dataset_name,
+            'file_id': file_id,
+        })
+
+
+@requires_identity
+def create_dataset_view(user):
+    """
+    Create a new dataset in cloned repository.
+
+    ---
+    get:
+        tags:
+        - "datasets"
+
+        summary: Create a new dataset with cached projects.
+        description: Create a new dataset with cached projects.
+
+        parameters:
+            - name: project_id
+              in: query
+              description: Project identifier.
+              schema:
+                type: string
+              required: true
+
+            - name: dataset_name
+              in: query
+              description: Dataset name.
+              schema:
+                type: string
+              required: true
+
+            - name: remote_name
+              in: query
+              description: Name of the remote to sync the project with.
+              schema:
+                type: string
+
+        responses:
+            result:
+                description: Contains `dataset` name of created dataset.
+
+            error:
+                description: |
+                  Contains `code` and `message` describing the error.
+
+    """
+    if request.method == 'GET':
+        dataset_name = request.args.get('dataset_name')
+        project_id = request.args.get('project_id')
+        pwd = validate_project_id(user, project_id)
+
+        remote_name = request.args.get('remote_name', 'origin')
+
+        if not pwd:
+            return jsonify(
+                error={
+                    'code': INVALID_PARAMS_ERROR_CODE,
+                    'message': 'invalid project_id argument',
+                }
+            )
+
+        cwd = os.getcwd()
+        os.chdir(str(pwd))
```

Comment:
use `LocalClient`

Reply:
No, we don't want to deviate implementations and how we use the core module. See above.